src/Build/BuildCheck/Infrastructure/AnalysisContext/AnalysisLoggingContext.cs

```diff
@@ -15,7 +15,11 @@ namespace Microsoft.Build.Experimental.BuildCheck;
 /// <summary>
 /// <see cref="IAnalysisContext"/> that uses <see cref="ILoggingService"/> to dispatch.
 /// </summary>
-internal class AnalysisLoggingContext : IAnalysisContext
+/// <remarks>
+/// Making this a record struct to avoid allocations (unless called through interface - which leads to boxing).
+/// This is wanted since this can be used in a hot path (of property reads and writes)
+/// </remarks>
+internal readonly record struct AnalysisLoggingContext : IAnalysisContext
```

Comment:
It's mainly to avoid (or better say postpone) allocations when passing data from build execution engine - untill it's decided whether any diagnostics will actually use the data.

Specifically here: https://github.com/dotnet/msbuild/pull/10237/files#diff-7de9562a624a633b107b17a69555c5ad62fa40481e12921281e1afd375d997d0R319

```
        public void ProcessPropertyRead(PropertyReadInfo propertyReadInfo)
            => _loggingService.BuildEngineDataRouter.ProcessPropertyRead(
                    propertyReadInfo,
                    new AnalysisLoggingContext(_loggingService, BuildEventContext));
```

Reply:
Oh I misunderstood the question  yeah no need to be a record, just struct - fixing..